apps/src/templates/VersionHistoryWithCommits.jsx

```diff
@@ -0,0 +1,341 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import {sources as sourcesApi, files as filesApi} from '../clientApi';
+import project from '@cdo/apps/code-studio/initApp/project';
+import firehoseClient from '@cdo/apps/lib/util/firehose';
+import * as utils from '../utils';
+import i18n from '@cdo/locale';
+import StylizedBaseDialog from '@cdo/apps/componentLibrary/StylizedBaseDialog';
+import Button from '@cdo/apps/templates/Button';
+import color from '@cdo/apps/util/color';
+/**
+ * A component for viewing project version history.
+ */
+export default class VersionHistoryWithCommits extends React.Component {
+  static propTypes = {
+    handleClearPuzzle: PropTypes.func.isRequired,
+    isProjectTemplateLevel: PropTypes.bool.isRequired,
+    useFilesApi: PropTypes.bool.isRequired,
+    onClose: PropTypes.func.isRequired,
+    isOpen: PropTypes.bool.isRequired
+  };
+
+  /**
+   * state:
+   * {
+   *   statusMessage: string,
+   *   versions: (null|{
+   *     lastModified: Date,
+   *     isLatest: boolean,
+   *     versionId: string,
+   *     comment: string
+   *   }[]),
+   *   showSpinner: boolean,
+   *   confirmingClearPuzzle: boolean,
+   *   isOpen: boolean
+   * }
+   */
+  constructor(props) {
+    super(props);
+    this.state = {
+      versions: null,
+      statusMessage: '',
+      showSpinner: true,
+      confirmingClearPuzzle: false,
+      isOpen: true
+    };
+
+    if (props.useFilesApi) {
```

Comment:
is there a case where we use the files api and would want to use this component?

Reply:
It looks like no? I traced this back to https://github.com/code-dot-org/code-dot-org/pull/11041 which has very little documentation and I think might've been dropped? I can remove this, just wasn't sure if it would break something in production

Reply:
But searching our code base, I don't think anything will break so I'll remove it. Looks like it's just some tech debt :(